profiling/src/profiling/mod.rs

```diff
@@ -292,6 +309,72 @@ impl TimeCollector {
         #[cfg(feature = "allocation_profiling")]
         let alloc_size_offset = sample_types.iter().position(|&x| x.r#type == "alloc-size");
 
+        // check if we have the IO sample types
+        #[cfg(all(target_os = "linux", feature = "io_profiling"))]
+        let socket_read_time_offset = sample_types
+            .iter()
+            .position(|&x| x.r#type == "socket-read-time");
+        #[cfg(all(target_os = "linux", feature = "io_profiling"))]
+        let socket_read_time_samples_offset = sample_types
+            .iter()
+            .position(|&x| x.r#type == "socket-read-time-samples");
+        #[cfg(all(target_os = "linux", feature = "io_profiling"))]
+        let socket_write_time_offset = sample_types
+            .iter()
+            .position(|&x| x.r#type == "socket-write-time");
```

Comment:
Perhaps we can infer that if we have the `socket-read-time`, that for everything after the offset just has to be incremented ... maybe in a future cleanup